src/planner/dart/ConfigurationToConfiguration_to_ConfigurationToTSR.cpp

```diff
@@ -0,0 +1,111 @@
+#include "aikido/planner/dart/ConfigurationToConfiguration_to_ConfigurationToTSR.hpp"
+#include <dart/dynamics/dynamics.hpp>
+#include "aikido/constraint/dart/InverseKinematicsSampleable.hpp"
+#include "aikido/constraint/dart/JointStateSpaceHelpers.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSaver.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+
+using aikido::constraint::dart::InverseKinematicsSampleable;
+using aikido::constraint::dart::TSR;
+using aikido::constraint::dart::createSampleableBounds;
+using aikido::statespace::dart::MetaSkeletonStateSaver;
+using aikido::statespace::dart::MetaSkeletonStateSpace;
+using ::dart::dynamics::InverseKinematics;
+
+namespace aikido {
+namespace planner {
+namespace dart {
+
+//==============================================================================
+ConfigurationToConfiguration_to_ConfigurationToTSR::
+    ConfigurationToConfiguration_to_ConfigurationToTSR(
+        std::shared_ptr<ConfigurationToConfigurationPlanner> planner)
+  : PlannerAdapter<ConfigurationToConfigurationPlanner,
+                   ConfigurationToTSRPlanner>(std::move(planner))
+{
+  // Do nothing
+}
+
+//==============================================================================
+trajectory::TrajectoryPtr
+ConfigurationToConfiguration_to_ConfigurationToTSR::plan(
+    const ConfigurationToTSR& problem, Planner::Result* result)
+{
+  // TODO: Check equality between state space of this planner and given problem.
+
+  // TODO: DART may be updated to check for single skeleton
+  if (mMetaSkeleton->getNumDofs() == 0)
+    throw std::invalid_argument("MetaSkeleton has 0 degrees of freedom.");
+
+  auto skeleton = mMetaSkeleton->getDof(0)->getSkeleton();
+  for (size_t i = 1; i < mMetaSkeleton->getNumDofs(); ++i)
+  {
+    if (mMetaSkeleton->getDof(i)->getSkeleton() != skeleton)
+      throw std::invalid_argument("MetaSkeleton has more than 1 skeleton.");
+  }
+
+  // Create an IK solver with MetaSkeleton DOFs
+  ::dart::dynamics::BodyNodePtr endEffectorBodyNode;
+  for (std::size_t i = 0; i < mMetaSkeleton->getNumBodyNodes(); ++i)
+  {
+    if (mMetaSkeleton->getBodyNode(i)->getName()
+        == problem.getEndEffectorBodyNode()->getName())
+    {
+      endEffectorBodyNode = mMetaSkeleton->getBodyNode(i);
+      break;
+    }
+  }
+  if (!endEffectorBodyNode)
+    throw std::invalid_argument(
+        "End-effector BodyNode not found in Planner's MetaSkeleton.");
+
+  auto ik = InverseKinematics::create(endEffectorBodyNode);
+  ik->setDofs(mMetaSkeleton->getDofs());
+
+  // Convert TSR constraint into IK constraint
+  InverseKinematicsSampleable ikSampleable(
+      mMetaSkeletonStateSpace,
+      mMetaSkeleton,
+      std::const_pointer_cast<TSR>(problem.getGoalTSR()),
```

Comment:
Nit: Could you make a note here saying this const-casting should be removed once `InverseKinematicsSampleable` is changed to take const constraints?